server/src/test/java/org/candlepin/model/ConsumerCuratorTest.java

```diff
@@ -314,10 +310,50 @@ public void testGetConsumersNullInput() {
         assertTrue(expected.contains(c2));
         assertTrue(expected.contains(c3));
 
-        List<Consumer> actual = consumerCurator.getConsumers(null).list();
+        Collection<Consumer> actual = consumerCurator.getConsumers(null);
         assertEquals(0, actual.size());
     }
 
+    @Test
+    public void testGetConsumerInputPartioning() {
+        int partitionBlockSize = 5;
+
+        ConsumerCurator curator = Mockito.spy(consumerCurator);
+        when(curator.getInBlockSize()).thenReturn(partitionBlockSize);
+
+        Map<String, Consumer> consumers = new HashMap<>();
```

Comment:
This test setup is quite noisy and since it is repeated in multiple tests we could help readability by extracting it to methods.

Reply:
I had considered it, but ended up not for the following reasons:

- Only two methods use it
- The point at which you pull common code out would only catch the consumer creation loop, having an input for the curator and output of the map. This then requires that the tests know the details of how the UUIDs are generated for them to work properly when generating the ID/UUID lists for method input, strongly coupling the two together.
- Getting *all* of the setup would require additional inputs of the "input" and "expected" sets so they can be populated for the call and verification steps. This basically ties them exactly to working in this one very specific way, and only serves to divide the code for the sake of dividing the code.

I get the DRY arguments, but I would argue that context switches to understand why a given setup is being performed, and convoluted method I/O doesn't make code easier to read in this case.